modules/govuk/manifests/apps/asset_manager.pp

```diff
@@ -88,7 +88,40 @@
       location ~ /raw/(.*) {
         internal;
         alias /var/apps/asset-manager/uploads/assets/$1;
-      }',
+      }
+
+      location = /cloud-storage-proxy/ {
+        # Prevent requests to this location from outside nginx
+        internal;
+
+        # Extract download host and URL from upstream HTTP headers
+        set $cloud_storage_host $upstream_http_x_cloud_storage_host;
+        set $download_url $upstream_http_x_cloud_storage_url;
+
+        # The X-CLOUD-STORAGE-URL header contains a signed URL for the asset on
+        # S3. The signature of this URL is based in part on the request headers
+        # set in the asset manager rails app at the time the URL is generated.
+        # The headers we send now must match otherwise nginx will not be
+        # allowed to make the request. Since this location block inherits
+        # `proxy_set_header` directives from previous levels[1], we explicitly
+        # set the Host so that the inherited headers are over-written.
+        #
+        # [1] http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_set_header
+        proxy_set_header Host $cloud_storage_host;
+
+        # Explicitly remove S3 HTTP headers
+        proxy_hide_header x-amz-id-2;
+        proxy_hide_header x-amz-request-id;
+        proxy_hide_header x-amz-version-id;
```

Comment:
Minor: I think this is fine for the moment, but if there was a way to use a regular expression (i.e. headers starting with `x-amz-`) to identify S3 headers that would be nicer since it would be more robust to S3 changes.